hasktorch/src/Torch/Random.hs

```diff
@@ -105,3 +131,12 @@ normal mean std size opts gen = generatorFactory (LibTorch.normal_ddlGo (realToF
 
 normal' :: Double -> Double -> [Int] -> Generator -> (Tensor,Generator)
 normal' mean std size gen = normal mean std size defaultOpts gen
+
+-- | Pure functional dropout
+dropout
+  :: Double -- ^ dropout probability
+  -> Maybe Generator -- ^ generator
+  -> Tensor -- ^ input
+  -> (Tensor,Generator) -- ^ output
+dropout p Nothing input = cast3 ATen.dropout_tdb input p False
```

Comment:
Hi @junjihashimoto, this looks great, thank you for working on this.
This line here is odd, though. I don't understand how this can return a `Generator` when `Nothing` is passed into it. Isn't `ATen.dropout_tdb` the one that returns `IO Tensor`?

Reply:
Thank you for your pointing it out. If the input generator is Nothing, the output generator should also return Nothing or just a Tensor(not the Tuple).